### PR TITLE
feat: require --reason on grouped destructive recovery verbs

### PR DESCRIPTION
**Pull request by JuaniRios**
## What
Implements [RAI-982](https://linear.app/makeitrain/issue/RAI-982) — Step 3 of [RAI-978](https://linear.app/makeitrain/issue/RAI-978). Makes `--reason` required on the new grouped destructive verbs.

## Why
A defaulted `--reason` ("Manually failed via CLI") is an audit-hostile record and violates the repo's no-silent-defaults rule. The reason is persisted as the audit trail for a manual intervention.

## How
Removes `default_value` from `--reason` on `transfer fail`, `transfer reconcile`, and `position release-hedge` (`position set` already required it). The hidden legacy flat commands (`fail-transfer`, `reconcile-usdc-transfer`) keep their defaults so existing runbooks/scripts still work. Note: `position release-hedge` and its alias `repair fail-pending-offchain-order` are the same clap variant, so the legacy alias now also requires `--reason` (it can't differ from the new name).

## Testing
Parse tests assert the new names error without `--reason` and the legacy flat name keeps its default. 148 `cli::` tests pass; clippy + fmt clean.

## Anything else
Part of epic [RAI-978](https://linear.app/makeitrain/issue/RAI-978).

## Summary by CodeRabbit

* **New Features**
  * Operator-intervention commands now require a non-empty `--reason`, preventing blank or missing reasons at the command line.

* **Bug Fixes**
  * Updated recovery and transfer workflows so grouped destructive actions no longer fall back to default reasons.
  * Preserved legacy default reasons for hidden compatibility commands, while keeping standalone commands working as expected.

* **Tests**
  * Added coverage for required reasons, blank-value rejection, and legacy command compatibility.

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/811?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#815** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/815?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#818](https://github.com/ST0x-Technology/st0x.liquidity/pull/818) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/818?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#935](https://github.com/ST0x-Technology/st0x.liquidity/pull/935) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/935?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#813** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/813?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#812** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/812?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#811** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/811?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/811?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#810** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#809** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/809?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#808** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/808?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#807** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/807?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#806** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/806?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#781** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/781?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#782](https://github.com/ST0x-Technology/st0x.liquidity/pull/782) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/782?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>add-to-gt-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ST0x-Technology?ref=merge-queue-instructions-in-stack-comment&prId=811).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by JuaniRios** [comment]
@coderabbitai full review

File: src/cli/mod.rs [hunk 1]
```diff
@@ -135,12 +135,9 @@ pub enum PositionRecoveryCommand {
...
         order_id: OffchainOrderId,
-        /// Reason to persist on the Position::FailOffChainOrder event
-        #[arg(
-            short = 'r',
-            long = "reason",
-            default_value = "Manually failed pending offchain order via CLI"
-        )]
+        /// Reason to persist on the Position::FailOffChainOrder event (required;
+        /// the audit record for this manual intervention)
+        #[arg(short = 'r', long = "reason", value_parser = parse_non_empty_reason)]
```
Comment (0xgleb) [inline src/cli/mod.rs:140]: should fix: `position set` (`PositionRecoveryCommand::Set`) is the sibling destructive verb but its `--reason` still has a plain `#[arg(short = 'r', long = "reason")]` with no `value_parser`, so `position set --reason "   "` persists a blank reason on `Position::ManualPositionAdjusted`. SPEC explicitly lists `set` among the event-emitting destructive verbs that must reject blank reasons, so this is exactly the audit-hostile record the PR exists to prevent. Mirror what you did here and add `value_parser = parse_non_empty_reason` to `Set.reason`.

File: src/cli/mod.rs [hunk 2]
```diff
@@ -197,6 +194,19 @@ fn parse_positive_shares(input: &str) -> Result<Positive<FractionalShares>, Stri
     Positive::new(shares).map_err(|err| format!("{err}"))
 }
 
+/// Rejects a blank `--reason` on event-emitting destructive verbs. clap already
+/// requires the flag be present once its default is removed, but a present-but-
+/// empty value (`--reason ""` or `--reason "   "`, easy to hit when a shell
+/// expands `--reason "$REASON"` to nothing) would persist the same audit-hostile
```
Comment (0xgleb) [inline src/cli/mod.rs:200]: minor: the non-blank invariant lives in clap attributes attached per-arg while the field stays a raw `String`, which inverts the newtype idiom and means every new persistence path has to remember to re-attach the validator. The leak is already real because `PositionRecoveryCommand::Set.reason` is a sibling audit reason that skipped it. A `Reason(String)` newtype with a parse impl rejecting `input.trim().is_empty()`, used for all three reason fields, would carry the invariant by construction so `Set` could not opt out. At minimum wire `parse_non_empty_reason` onto `Set.reason` for now.

File: src/cli/mod.rs [hunk 3]
```diff
@@ -2336,6 +2344,140 @@ mod tests {
...
+            clap::error::ErrorKind::MissingRequiredArgument
+        );
+        assert!(
+            error.to_string().contains("reason"),
+            "the missing argument must be --reason, got: {error}",
+        );
+    }
+
+    #[test]
+    fn transfer_fail_rejects_blank_reason() {
```
Comment (0xgleb) [inline src/cli/mod.rs:2387]: minor: there are blank-reason tests for `transfer fail` and `position release-hedge` but none for `position set`. Once `Set.reason` gets `parse_non_empty_reason`, add a `position_set_rejects_blank_reason` test in the same shape (`position set --symbol MSTR --zero --reason "   "`, asserting `ErrorKind::ValueValidation` and a "must not be blank" message) so a future removal of the guard fails the suite instead of silently restoring the blank-reason behavior.

File: SPEC.md [hunk 4]
```diff
@@ -3119,11 +3119,13 @@ effect rather than a generic intent:
...
-  The rename step carries those defaults onto the renamed forms unchanged
-  (`fail-pending-offchain-order` is a clap alias sharing its canonical command's
-  arguments, and the flat `fail-transfer` / `reconcile-usdc-transfer` variants
-  keep their legacy shape for compatibility); the next step of the unification
-  epic removes the defaults everywhere.
+  The rename step carried those defaults onto the renamed forms unchanged; the
+  defaults-removal step then strips them from the grouped verbs and, with them,
+  from `fail-pending-offchain-order` (a clap alias shares its canonical
+  command's arguments, so the alias cannot keep a default its canonical form
+  lost). Only the hidden flat `fail-transfer` / `reconcile-usdc-transfer`
```
Comment (0xgleb) [inline SPEC.md]: minor: this "Only the hidden flat `fail-transfer` / `reconcile-usdc-transfer` variants retain their defaulted reasons" claim is inaccurate. `fail-usdc-transfer` (`FailUsdcTransfer`) also keeps `default_value = "Manually failed via CLI"`, and it is not hidden (no `#[command(hide = true)]`), so both the "Only" and "hidden" qualifiers misdescribe it. Since SPEC is the source of truth, either add `fail-usdc-transfer` to the enumeration and drop "hidden", or scope the sentence to the fail/reconcile grouped-verb unification so it does not read as exhaustive.

**Review by 0xgleb — CHANGES_REQUESTED** [review]

File: src/cli/mod.rs (hunk 1 above)
Comment (JuaniRios) [reply src/cli/mod.rs:140]: Wired `parse_non_empty_reason` onto `Set.reason` — it was the only event-emitting destructive verb missing it. ef74d966

File: src/cli/mod.rs (hunk 2 above)
Comment (JuaniRios) [reply src/cli/mod.rs:200]: Agreed the newtype is the more robust encoding, but it ripples into three handler signatures and is out of scope here; the per-arg validator on `Set.reason` closes the actual leak you flagged. Worth a follow-up rather than expanding this PR. ef74d966

File: src/cli/mod.rs (hunk 3 above)
Comment (JuaniRios) [reply src/cli/mod.rs:2387]: Added `position_set_rejects_blank_reason` (asserts `ValueValidation` + 'must not be blank'). ef74d966

File: SPEC.md (hunk 4 above)
Comment (JuaniRios) [reply SPEC.md]: Fixed the SPEC claim — `fail-usdc-transfer` also carries a defaulted reason and is neither hidden nor has a grouped equivalent. (#812 then rewrites this passage for the post-deprecation state.) ef74d966

**Review by 0xgleb — APPROVED** [review]
